simulator/src/main/java/com/hazelcast/simulator/worker/WorkerCommandRequestProcessor.java

```diff
@@ -145,7 +145,15 @@ private void doProcess(long id, Command command) throws Exception {
         }
 
         private boolean process(IsPhaseCompletedCommand command) {
-            return !commands.containsKey(command.testId);
+            TestPhase testPhase = testPhases.get(command.testId);
```

Comment:
This is a change in the behavior. If we are asking for a phase, but another phase is still running, we throw an `IllegalStateException`. This is new, but I think it may help us to gather more information in case of another race during a test run.